core/dbt/include/global_project/macros/materializations/tests/test.sql

```diff
@@ -1,8 +1,33 @@
 {%- materialization test, default -%}
 
   {% set relations = [] %}
+    {% set relation_type = config.get('strategy') %}
 
-  {% if should_store_failures() %}
+    {% if relation_type in ['view', 'table'] %}
+
+        {% set identifier = model['alias'] %}
+        {% set existing_relation = adapter.get_relation(database=database, schema=schema, identifier=identifier) %}
+        {% set target_relation = api.Relation.create(
+            database=database, schema=schema, identifier=identifier, type=relation_type
+        ) %}
+
+        {% call statement(auto_begin=True) %}
+            {% if existing_relation %}
+                {{ get_replace_sql(existing_relation, target_relation, sql) }}
+            {% else %}
+                {{ get_create_sql(target_relation, sql) }}
+            {% endif %}
+        {% endcall %}
+
+        {% do relations.append(target_relation) %}
+
+        {% set main_sql %}
+            select * from {{ target_relation }}
+        {% endset %}
+
+        {{ adapter.commit() }}
+
+  {% elif should_store_failures() %}
```

Comment:
here's my read on the logic

1. if `relation_type` is configured, then depending on the argument materialize a table or view
2. else, if `should_store_failures` do all that jazz.

the first thing reads to me effectively as a sub-materialization, that is independent of that. so might i be able to pass `should_store_failures() and relation_type='table'`? what happens then?

I'm down with a deprecation in the future, but in the meantime might be deprecate the implementation and make `config.get('store_failures')` imply `relation_type='table'`? then there's only one code path for creating a table and storing test results?

Reply:
> so might i be able to pass should_store_failures() and relation_type='table'? what happens then?

It will create it as a table. This is the same as omitting the 'table' piece entirely.

> but in the meantime might be deprecate the implementation and make config.get('store_failures') imply relation_type='table'?

That's the conclusion that @colin-rogers-dbt and I reached as well. That alters the feature request slightly, but it makes much more intuitive sense. The feature request is now effectively:

I know `store_failures` creates the results as as table, but I want to make it create the results as a view.

Instead of implementing a feature alongside `store_failures`, we're augmenting `stored_failures` itself.